backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java

```diff
@@ -0,0 +1,117 @@
+import org.slf4j.LoggerFactory;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+import com.github.dockerjava.api.DockerClient;
+
+public class DockerCassandra {
+
+    private static final Logger logger = LoggerFactory.getLogger(DockerCassandra.class);
+
+    private static final int CASSANDRA_PORT = 9042;
+    private static final String CASSANDRA_CONFIG_DIR = "$CASSANDRA_CONFIG";
+    private static final String CASSANDRA_YAML = CASSANDRA_CONFIG_DIR + "/cassandra.yaml";
+    private static final String CASSANDRA_ENV = CASSANDRA_CONFIG_DIR + "/cassandra-env.sh";
+    private static final String JVM_OPTIONS = CASSANDRA_CONFIG_DIR + "/jvm.options";
+
+    private final GenericContainer<?> cassandraContainer;
+    private final DockerClient client;
+
+    @SuppressWarnings("resource")
+    public DockerCassandra() {
+        client = DockerClientFactory.instance().client();
+        boolean deleteOnExit = false;
+        cassandraContainer = new GenericContainer<>(
+            new ImageFromDockerfile("cassandra_2_2_12", deleteOnExit)
+                .withDockerfileFromBuilder(builder ->
+                    builder
+                        .from("cassandra:2.2.12")
+                        .env("ENV CASSANDRA_CONFIG", "/etc/cassandra")
+                        //avoiding token range computation helps starting faster
+                        .run("echo \"JVM_OPTS=\\\"\\$JVM_OPTS -Dcassandra.initial_token=0\\\"\" >> " + CASSANDRA_ENV)
+                        .run("sed -i -e \"s/num_tokens/\\#num_tokens/\" " + CASSANDRA_YAML)
+                        //don't wait for other nodes communication to happen
+                        .run("echo \"JVM_OPTS=\\\"\\$JVM_OPTS -Dcassandra.skip_wait_for_gossip_to_settle=0\\\"\" >> " + CASSANDRA_ENV)
+                        //auto_bootstrap should be useless when no existing data
+                        .run("echo auto_bootstrap: false >> " + CASSANDRA_YAML)
+                        .run("echo \"-Xms1500M\" >> " + JVM_OPTIONS)
+                        .run("echo \"-Xmx1500M\" >> " + JVM_OPTIONS)
+                        // disable assertions (modest performance benefit)
+                        .run("sed -i -e 's/JVM_OPTS=\"$JVM_OPTS -ea\"/JVM_OPTS=\"$JVM_OPTS -da\"/' " + CASSANDRA_ENV)
+                        // use caches for keys & rows
+                        .run("sed -i -e \"s/key_cache_size_in_mb:/key_cache_size_in_mb: 256/\" " + CASSANDRA_YAML)
+                        .run("sed -i -e \"s/row_cache_size_in_mb: 0/row_cache_size_in_mb: 512/\" " + CASSANDRA_YAML)
+                        .build()))
+            .withCreateContainerCmdModifier(cmd -> cmd.getHostConfig().withTmpFs(ImmutableMap.of("/var/lib/cassandra", "rw,noexec,nosuid,size=1g")))
```

Comment:
I already have hard time playing cassandra test with current memory requirements, I really don't want to increase them

Reply:
what about keeping the strategy of playing a limited number of tests on a given container ? or maybe monitor /var/lib/cassandra usage and restart the container when it's almost full ?